scripts/administration/id_generator.py

```diff
@@ -0,0 +1,68 @@
+import argparse
+import json
+import os
+
+from domain.core.cpm_system_id import ProductId
+
+ID_FILE_MAP = {
+    "product": f"{os.getcwd()}/src/layers/domain/core/cpm_system_id/generated_ids/product_ids.json",
+}
+
+ID_CLASS_MAP = {
+    "product": ProductId,
+}
+
+
+def save_to_set(id, ids):
+    if id in ids:
+        return False
+
+    ids.add(id)
+    return ids
+
+
+def generator(id_type="product"):
+    id_class = ID_CLASS_MAP.get(id_type)
+    generator = id_class.create()
+    return generator.__root__
+
+
+def open_file(id_type="product"):
+    if os.path.exists(ID_FILE_MAP.get(id_type)):
+        with open(ID_FILE_MAP.get(id_type), "r") as file:
+            return set(json.load(file))
+    return set()
+
+
+def save_file(ids, id_type="product"):
+    with open(ID_FILE_MAP.get(id_type), "w") as file:
+        json.dump(list(ids), file)
+
+
+def bulk_generator(id_count=1, id_type="product"):
+    ids = open_file(id_type=id_type)
+    for _ in range(id_count):
+        saved = False
+        while not saved:
+            generated_id = generator()
+            saved = save_to_set(generated_id, ids)
+        ids = saved
```

Comment:
maybe to make more concise:

```suggestion
    initial_len = len(ids)
    while len(ids) - initial_len < id_count:
        ids.add(generator())
```